library/systemd/src/modules/systemd_service.rb

```diff
@@ -174,18 +174,118 @@ def restart
 
       # Returns socket associated with service or nil if there is no such socket
       def socket
+        return @socket if @socket
```

Comment:
this will work only if socket is not found. But I expect that check for triggered_by is quick enough. What I do not like is that returned socket is cached for long time and it still has old status. Maybe at least refresh_properties should be called?

Reply:
I just wanted to avoid calling systemctl several times. For instance, if we do something like:

```ruby
service.socket.disable if service.socket?
```

We would be calling `#socket` twice. Having some kind of `reload` might help (which could be implemented calling `#find` again).

Reply:
If socket is not found in a first call, it will try to find it again in subsequent calls. Does it make sense? I don't know.